src/coprocessor/endpoint.rs

```diff
@@ -65,14 +65,36 @@ pub struct Host {
     sched: Scheduler<Task>,
     reqs: HashMap<u64, Vec<RequestTask>>,
     last_req_id: u64,
-    pool: ThreadPool<FifoQueue<u64>, u64>,
-    low_priority_pool: ThreadPool<FifoQueue<u64>, u64>,
-    high_priority_pool: ThreadPool<FifoQueue<u64>, u64>,
+    pool: ThreadPool<FifoQueue<u64, DummyContext>, u64, DummyContextFactory, DummyContext>,
+    low_priority_pool:
+        ThreadPool<FifoQueue<u64, DummyContext>, u64, DummyContextFactory, DummyContext>,
+    high_priority_pool:
+        ThreadPool<FifoQueue<u64, DummyContext>, u64, DummyContextFactory, DummyContext>,
     max_running_task_count: usize,
 }
 
+// TODO: remove these dummy structures.
+#[derive(Clone)]
+struct DummyContext {}
+
+unsafe impl Send for DummyContext {}
+
+impl Context for DummyContext {
+    fn on_task_started(&self, _: &mut Self) {}
```

Comment:
using &mut self may be better? so we can update the context directly, it is thread-safe.